mms/model_loader.py

```diff
@@ -7,53 +7,180 @@
 
-from mms.log import get_logger
-
-logger = get_logger()
+from mms.service import Service
+from mms.model_service.mxnet_model_service import SingleNodeService
+from mms.log import log_error
 
 MANIFEST_FILENAME = 'MANIFEST.json'
 
+class ModelLoaderFactory(object):
+    """
+    ModelLoaderFactory
+    """
+
+    @staticmethod
+    def get_model_loader(service_type):
+        if service_type == "mms":
+            return MmsModelLoader()
+        elif service_type == "legacy_mms":
+            return LegacyModelLoader()
+        else:
+            raise ValueError("Unknown model loader type: {}".format(service_type))
+
 
 class ModelLoader(object):
     """
-    Model Loader
+    Base Model Loader class.
     """
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def load(self, model_name, model_dir, handler, gpu_id, batch_size):
+        """
+        Load model from file.
+
+        :param model_name:
+        :param model_dir:
+        :param handler:
+        :param gpu_id:
+        :param batch_size:
+        :return: Model
+        """
+        pass
+
     @staticmethod
-    def load(model_dir, handler):
+    def list_model_services(module, parent_class=None):
+        """
+        Parse user defined module to get all model service classes in it.
+
+        :param module:
+        :param parent_class:
+        :return: List of model service class definitions
         """
-        Load models
 
-        Parameters
-        ----------
-        model_dir : str
-            Model path
-        handler : str
-            Model entry point service handler file name
+        # Parsing the module to get all defined classes
+        classes = [cls[1] for cls in inspect.getmembers(module, lambda member: inspect.isclass(member) and
+                                                        member.__module__ == module.__name__)]
+        print(classes)
+        # filter classes that is subclass of parent_class
+        if parent_class is not None:
+            return [c for c in classes if issubclass(c, parent_class)]
+        else:
+            return classes
 
-        Returns
-        ----------
-        map
-            Model manifest
+
+class MmsModelLoader(ModelLoader):
+    """
+    MMS 1.0 Model Loader
+    """
+
+    def load(self, model_name, model_dir, handler, gpu_id, batch_size):
         """
-        manifest_file = os.path.join(model_dir, "MANIFEST.legacy")
-        if os.path.isfile(manifest_file):
-            manifest = json.load(open(manifest_file))
+        Load MMS 1.0 model from file.
+
+        :param model_name:
+        :param model_dir:
+        :param handler:
+        :param gpu_id:
+        :param batch_size:
+        :return:
+        """
+        manifest_file = os.path.join(model_dir, MANIFEST_FILENAME)
+        manifest = None
+        if os.path.exists(manifest_file):
+            with open(manifest_file) as f:
+                manifest = json.load(f)
+
+        temp = handler.split(':', 1)
```

Comment:
this should be split(':', 2)

Reply:
No this should be split one, as we expect only one ':' it will split on the first occurrence of ':' only